python/desc/imsim/bleed_trails.py

```diff
@@ -0,0 +1,175 @@
+
+    Parameters
+    ----------
+    eimage: lsst.afw.image.ImageF
+        The "eimage" containing the image data in units of electrons per
+        pixel.  This is the image prior to electronic readout (i.e.,
+        conversion to ADU, addition of bias, dark current, crosstalk, etc.).
+        For LSST CCDs, the eimages have parallel transfer directions along
+        the x-axis, hence channels correspond to rows in eimages.
+    full_well: int
+        The pixel full well/saturation value in electrons.
+    midline_stop: bool [True]
+        Flag to treat the midline of the sensor as a bleed stop.
+
+    Returns
+    -------
+    afw.image.ImageF: This is the input eimage object with the charge
+        bleeding applied.
+    """
+    channels = find_channels_with_saturation(eimage, full_well)
+    imarr = eimage.getArray()
+    for ypix in channels:
+        if midline_stop:
+            xmid = imarr.shape[1]//2
+            imarr[ypix, :xmid] = bleed_channel(imarr[ypix, :xmid], full_well)
+            imarr[ypix, xmid:] = bleed_channel(imarr[ypix, xmid:], full_well)
+        else:
+            imarr[ypix, :] = bleed_channel(imarr[ypix, :], full_well)
+    return eimage
+
+
+def find_channels_with_saturation(eimage, full_well):
+    """
+    Find all channels in an eimage that have pixel values above full
+    well.
+
+    Parameters
+    ----------
+    eimage: lsst.afw.image.ImageF
+        The "eimage" containing the image data in units of electrons per
+        pixel.  This is the image prior to electronic readout (i.e.,
+        conversion to ADU, addition of bias, dark current, crosstalk, etc.).
+        For LSST CCDs, the eimages have parallel transfer directions along
+        the x-axis, hence channels correspond to rows in eimages.
+    full_well: int
+        The pixel full well/saturation value in electrons.
+
+    Returns
+    -------
+    set of ints:  The y-indices of the channels with saturated pixels.
+    """
+    threshold = afw_detect.Threshold(full_well)
+    fp_set = afw_detect.FootprintSet(eimage, threshold)
+    channels = []
+    for fp in fp_set.getFootprints():
+        for span in fp.spans:
+            channels.append(span.getY())
+    return set(channels)
+
+
+def bleed_channel(channel, full_well):
+    """
+    Redistribute charge along a channel for all pixels above full well.
+
+    Parameters
+    ----------
+    channel: numpy.array of pixel values
+        1D array of pixel values in units of electrons.
+    full_well: int
+        The pixel full well/saturation value in electrons.
+
+    Returns
+    -------
+
+    numpy.array of pixel values: The channel of pixel data with
+       bleeding applied.  This is a new object, i.e., the input
+       numpy.array is unaltered.
+    """
+    # Construct an afw_image.ImageF to contain the channel data
+    # to enable the use of afw_detect to find the above full-well
+    # footprints.
+    nx = len(channel)
+    image = afw_image.ImageF(nx, 1)
```

Comment:
Is this guaranteed to always be initialized to zero so when you add the channel data to it, it will be OK?

Reply:
Yes, that's the default value:  https://github.com/lsst/afw/blob/master/include/lsst/afw/image/Image.h#L95